src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -199,31 +220,228 @@ private Future<String> fetchToken(JsonObject payload, String tenant,
+    var exp = Instant.parse(refreshTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    var sb = new StringBuilder("refreshToken=");
+    sb.append(refreshToken);
+
+    // Path indicates the path that must exist in the requested URL for the client to send the
+    // Cookie header.
+    sb.append("; HttpOnly; Secure; Path=/authn; ");
+
+    // The Max-Age attribute is the number seconds until the token and cookie expires. It also ensures
+    // that the cookie will persist between browser tab sessions.
+    sb.append("Max-Age=");
+    sb.append(ttlSeconds);
+
+    return sb.toString();
+  }
+
+  private String accessTokenCookie(String accessToken, String accessTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(accessTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    var sb = new StringBuilder("accessToken=");
+    sb.append(accessToken);
+
+    // Path indicates the path that must exist in the requested URL for the client to send the
+    // Cookie header.
+    sb.append("; HttpOnly; Secure; ");
+
+    // The Max-Age attribute is the number seconds until the token and cookie expires. It also ensures
+    // that the cookie will persist between browser tab sessions.
+    sb.append("Max-Age=");
+    sb.append(ttlSeconds);
+
+    return sb.toString();
+  }
+
+  private Response tokenResponse(JsonObject fetchTokenFuture) {
+    String accessToken = fetchTokenFuture.getString(ACCESS_TOKEN);
+    String refreshToken = fetchTokenFuture.getString(REFRESH_TOKEN);
+    String accessTokenExpiration = fetchTokenFuture.getString(ACCESS_TOKEN_EXPIRATION);
+    String refreshTokenExpiration = fetchTokenFuture.getString(REFRESH_TOKEN_EXPIRATION);
+    // Use the ResponseBuilder rather than RMB-generated code. We need to do this because
+    // RMB generated-code does not allow multiple headers with the same key -- which is what we need
+    // here.
+    var body = new JsonObject()
+        .put(ACCESS_TOKEN_EXPIRATION, accessTokenExpiration)
+        .put(REFRESH_TOKEN_EXPIRATION, refreshTokenExpiration)
+        .toString();
+    return Response.status(201)
+        .header(SET_COOKIE, refreshTokenCookie(refreshToken, refreshTokenExpiration))
+        .header(SET_COOKIE, accessTokenCookie(accessToken, accessTokenExpiration))
+        .type(MediaType.APPLICATION_JSON)
+        .entity(body)
+        .build();
+  }
+
+  private Response logoutResponse() {
+    // Use the ResponseBuilder rather than RMB-generated code. We need to do this because
+    // RMB generated-code does not allow multiple headers with the same key -- which is what we need
+    // here.
+    return Response.status(204)
+        .header(SET_COOKIE, REFRESH_TOKEN + "=")
+        .header(SET_COOKIE, ACCESS_TOKEN + "=")
```

Comment:
Instead of setting the cookie to an empty string it should be deleted.
https://stackoverflow.com/questions/20320549/how-can-you-delete-a-cookie-in-an-http-response
https://stackoverflow.com/questions/5285940/correct-way-to-delete-cookies-server-side

Reply:
Revised code to set `Expires` to time in past. I was a bit unsure about what to do with the cookie name/value. I left it setting the value to an empty string _and_ set `Expires` to time in the past. The IETF RFC _seems_ to indicate that UAs should match on the `Domain` and the `Path` only but I'm a bit suspicious of that. How would you delete one cookie for a given domain yet keep another?

Reply:
Yes, the name is required, the value is ignored and can be empty. Example from rfc6265 how to delete a cookie:
`Set-Cookie: lang=; Expires=Sun, 06 Nov 1994 08:49:37 GMT`